code/game/objects/structures/mannequin.dm

```diff
@@ -0,0 +1,1386 @@
+
+/////////////////////////////////////////////////////////MANNEQUINS//////////////////////////////////////////////////////////
+//Basically statues that you can dress up.
+
+/obj/structure/mannequin
+	name = "human marble mannequin"
+	desc = "You almost feel like it's going to come alive any second."
+	icon = 'icons/obj/mannequin.dmi'
+	icon_state="mannequin_marble_human"
+	density = 1
+	anchored = 1
+	plane = ABOVE_HUMAN_PLANE
+	layer = VEHICLE_LAYER
+	var/datum/species/species
+	var/species_type = /datum/species/human //TODO: mannequin for other races
+	var/fat = 0
+	var/bulky = 0
+	var/list/clothing = list()
+	var/list/obj_overlays = list()
+	var/list/obj/item/held_items = list(null, null)
+	var/clothing_offset_x = 0
+	var/clothing_offset_y = 3
+	var/health = 90
+	var/maxHealth = 90
+
+	//for mappers, with love
+	var/mapping_uniform = null
+	var/mapping_shoes = null
+	var/mapping_gloves = null
+	var/mapping_ears = null
+	var/mapping_suit = null
+	var/mapping_glasses = null
+	var/mapping_hat = null
+	var/mapping_belt = null
+	var/mapping_mask = null
+	var/mapping_back = null
+	var/mapping_id = null
+
+	var/mapping_hand_right = null
+	var/mapping_hand_left = null
+
+/obj/structure/mannequin/New()
+	..()
+
+	species = new species_type()
+
+	clothing = list(
+		SLOT_MANNEQUIN_ICLOTHING,
+		SLOT_MANNEQUIN_FEET,
+		SLOT_MANNEQUIN_GLOVES,
+		SLOT_MANNEQUIN_EARS,
+		SLOT_MANNEQUIN_OCLOTHING,
+		SLOT_MANNEQUIN_EYES,
+		SLOT_MANNEQUIN_BELT,
+		SLOT_MANNEQUIN_MASK,
+		SLOT_MANNEQUIN_HEAD,
+		SLOT_MANNEQUIN_BACK,
+		SLOT_MANNEQUIN_ID,
+		)
+	obj_overlays = list(
+		MANNEQUIN_UNIFORM_LAYER,
+		MANNEQUIN_SHOES_LAYER,
+		MANNEQUIN_GLOVES_LAYER,
+		MANNEQUIN_EARS_LAYER,
+		MANNEQUIN_SUIT_LAYER,
+		MANNEQUIN_GLASSES_LAYER,
+		MANNEQUIN_BELT_LAYER,
+		MANNEQUIN_FACEMASK_LAYER,
+		MANNEQUIN_HEAD_LAYER,
+		MANNEQUIN_BACK_LAYER,
+		MANNEQUIN_ID_LAYER,
+		)
+
+	for(var/i in 1 to held_items.len)
+		obj_overlays |= "[MANNEQUIN_HAND_LAYER]-[i]"
+
+	obj_overlays[MANNEQUIN_UNIFORM_LAYER]	= getFromPool(/obj/Overlays/uniform_layer)
+	obj_overlays[MANNEQUIN_ID_LAYER]		= getFromPool(/obj/Overlays/id_layer)
+	obj_overlays[MANNEQUIN_SHOES_LAYER]		= getFromPool(/obj/Overlays/shoes_layer)
+	obj_overlays[MANNEQUIN_GLOVES_LAYER]	= getFromPool(/obj/Overlays/gloves_layer)
+	obj_overlays[MANNEQUIN_EARS_LAYER]		= getFromPool(/obj/Overlays/ears_layer)
+	obj_overlays[MANNEQUIN_SUIT_LAYER]		= getFromPool(/obj/Overlays/suit_layer)
+	obj_overlays[MANNEQUIN_GLASSES_LAYER]	= getFromPool(/obj/Overlays/glasses_layer)
+	obj_overlays[MANNEQUIN_BELT_LAYER]		= getFromPool(/obj/Overlays/belt_layer)
+	obj_overlays[MANNEQUIN_BACK_LAYER]		= getFromPool(/obj/Overlays/back_layer)
+	obj_overlays[MANNEQUIN_FACEMASK_LAYER]	= getFromPool(/obj/Overlays/facemask_layer)
+	obj_overlays[MANNEQUIN_HEAD_LAYER]		= getFromPool(/obj/Overlays/head_layer)
+
+	checkMappingWear()
+
+
+/obj/structure/mannequin/Destroy()
+	new /obj/effect/decal/cleanable/dirt(loc)
```

Comment:
I'm really against doing stuff like this in destroy, because it gets called from too many places.

Should an admin deleting a mannequin cause dirt to drop? No.